imsim/lsst_image.py

```diff
@@ -266,9 +273,25 @@ def addNoise(self, image, config, base, image_num, obj_num, current_var, logger)
             logger.info("Applying vignetting according to radial spline model.")
             radii = Vignetting.get_pixel_radii(camera[det_name])
             sky.array[:] *= self.vignetting.apply_to_radii(radii)
-
+            
+        if self.apply_fringing:
+            # Use serial number as random seed number to make sure the height map 
+            # for the same sensor is the same for different exposures.
+            camera = get_camera(self.camera_name)
+            det_name = base['det_name']
+            serial_number = camera[det_name].getSerial()
+            # Only apply fringing to e2v sensors.
+            if serial_number[:3] == 'E2V':
+                ccd_fringing = CCD_Fringing(true_center=image.wcs.toWorld(image.true_center),
+                                            boresight=self.boresight,
+                                            seed=int(serial_number[-3:]), spatial_vary=True)
```

Comment:
This is ok, but maybe fragile.  If they change the structure of the string, this might start failing.  More robust is `seed=hash(serial_number)`.

Reply:
Got it.  I am thinking the line getting the sensor type `if serial_number[:3] == 'E2V'` maybe also fail? Is there any better options for that one?

Reply:
Mm.  That's also maybe fragile.  But less so.  It's probably unlikely they'll change the first three characters to something else.  But I could imaging them wanting to add letters at the end or underscores somewhere.  Who knows.  hash will convert any string to an integer.